shell/src/main/java/alluxio/cli/UnderFileSystemContractTest.java

```diff
@@ -111,42 +106,47 @@ private void runS3Operations() throws Exception {
     mUfs = UnderFileSystem.Factory.create(mUfsPath, getUfsConf());
 
     String testDir = createTestDirectory();
-    loadAndRunTests(new S3ASpecificOperations(testDir, mUfs, mConf), testDir);
+    return loadAndRunTests(new S3ASpecificOperations(testDir, mUfs, mConf), testDir);
   }
 
   /**
    * Loads and runs the tests in the target operations class.
    *
    * @param operations the class that contains the tests to run
    * @param testDir the test directory to run tests against
+   * @return the number of failed tests
    */
-  private void loadAndRunTests(Object operations, String testDir) throws Exception {
+  private int loadAndRunTests(Object operations, String testDir) throws Exception {
+    int failedTestCnt = 0;
     try {
       Class classToRun = operations.getClass();
-      Field[] fields = classToRun.getDeclaredFields();
-      for (Field field : fields) {
-        field.setAccessible(true);
-      }
       Method[] tests = classToRun.getDeclaredMethods();
       for (Method test : tests) {
         String testName = test.getName();
         if (testName.endsWith("Test")) {
           System.out.printf("Running test: %s...", testName);
+          boolean passed = false;
           try {
             test.invoke(operations);
-          } catch (InvocationTargetException e) {
+            passed = true;
+            cleanupUfs(testDir);
+          } catch (Exception e) {
             if (mUfs.getUnderFSType().equals(S3_IDENTIFIER)) {
               logRelatedS3Operations(test);
             }
-            throw new IOException(e.getTargetException());
+            System.err.format("Test %s.%s aborted%n%s", test.getClass(), test.getName(), e);
+          } finally {
+            RunTestUtils.printPassInfo(passed);
```

Comment:
does this print something different based on whether `passed` is true or false?

Reply:
I renamed `CliUtils` to `RunTestUtils`, as `CliUtils` contains only util methods local to `runUfsTests`, instead of general to all other cli classes.

`RunTestUtils.printPassInfo()` corresponds to 
https://github.com/Alluxio/alluxio/blob/c4eba4706750ffd7ad00c486af9454b4342bf914/examples/src/main/java/alluxio/cli/CliUtils.java#L34